packages/drivers/driver-base/src/documentDeltaConnection.ts

```diff
@@ -595,8 +595,17 @@ export class DocumentDeltaConnection
 			// Socket can be disconnected while waiting for Fluid protocol messages
 			// (connect_document_error / connect_document_success), as well as before DeltaManager
 			// had a chance to register its handlers.
-			this.addTrackedListener("disconnect", (reason) => {
+			this.addTrackedListener("disconnect", (reason, details) => {
 				const err = this.createErrorObject("disconnect", reason);
+				this.logger.sendTelemetryEvent(
```

Comment:
I agree (see my other comment) however, I'm identifying a small discrepancy between what's being actually sent (websocket code) vs what I was expecting (http code). I would like to first bake this and observe it in the wild and if it's stable, we can move forward with API changes, as to make this properly, we'd need an API change to `createErrorObject` which is a bit more expensive.

Reply:
Tangent (posting so it doesn't only live in my head, in case we make changes to `createErrorObject`): last time I was looking into it to improve handling of SSL cert errors, I noticed that due to the way our APIs are structured in that space, in certain cases (can't recall if the websocket, or the regular http request one) we throw a generic network error without details of the low-level issue (e.g. invalid SSL cert) and from then on the best we can do is something like a `error.message.startsWith("Network:")` check but cannot tell exactly what happened. So if we get to refactoring `createErrorObject`, happy to get involved with the details of that scenario (which I'll need to extract from ADO items 😄 ).